src/download-artifact.ts

```diff
@@ -131,6 +139,16 @@ async function run(): Promise<void> {
   core.info('Download artifact has finished successfully')
 }
 
-run().catch(err =>
-  core.setFailed(`Unable to download artifact(s): ${err.message}`)
-)
+{
+  const flags = {warnOnFailure: false}
+  run(flags).catch(err => {
+    if (flags.warnOnFailure) {
+      core.setOutput(
+        'failure',
+        `Unable to download artifact(s): ${err.message}`
+      )
```

Comment:
How about using a warning annotation instead? https://github.com/actions/toolkit/tree/main/packages/core#annotations

Reply:
No. Because I can't as a consuming action or a consuming workflow retrieve that without extra permissions and fishing through the logs.

If I'm the consuming action/workflow, I can use the output and _then_ produce my own annotation _if_ I decide it's relevant to my use case.

Reply:
For demonstration purposes, I could include that in the test.yml to show that it works...